vtex-apps/product-list-context
# Fix/position ordering

#### What problem is this solving?

 The position field should be ordered and incremental, reflecting the product’s position in that list of products. Currently, the “position” field is filled in a repeatable way. See the video that shows the current scenario.

![List Positio](https://user-images.githubusercontent.com/12702016/102388577-b9447f00-3fb0-11eb-9d2c-49ef1e826e10.gif)

Instead of “1 2 3 4 1 2 3 4 1 2 3 4”, we should have “1 2 3 4 5 6 7 8 9 10 11 12”.

Task https://vtex-dev.atlassian.net/jira/software/projects/SF/boards/105?assignee=5d794d62656e2a0dbab2e04d&selectedIssue=SF-595

I don't think this solution is a 100% guarantee to always reflect the correct position for all the rendered products, as we're depending on the size of the `sentIds` set to get the position of a product within the list context. In other words, we assume that the send impression call will always be called in the correct order.

#### How to test it?



Compare the following flow between 

https://kiwi--storecomponents.myvtex.com/apparel---accessories/

and

https://storecomponents.myvtex.com/apparel---accessories/

1) wait page to load
2) open console and type `dataLayer`
3) Scroll up and down to get some products on the viewport
4) type `dataLayer` again
5) look for `productImpression` events

The `kiwi` workspace should have incremental positions, while the `master` workspace always resets to 0, 1, 2, 3

#### Screenshots or example usage:

![image](https://user-images.githubusercontent.com/12702016/102388084-06742100-3fb0-11eb-95ce-119e52ba17fc.png)


#### Describe alternatives you've considered, if any.



#### How does this PR make you feel? [:link:](http://giphy.com/)



![](https://media.giphy.com/media/l2Je7a6s6zNKXPD1e/giphy.gif)
